### ISSUE TITLE
A blank page is added when placing a full page bg image 

### ISSUE THREAD

**Issue by mattyx96** [dompdf/dompdf#1994]
I tried to add a background image to my document but if I add a full size background image (A4  width: 210mm;height: 297mm;) it adds me a blank page. If I try instead with height: 290mm the blank image doesn't appear but a small blank space is left.

```
<style>
        
        .background {
            width: 210mm;
            height: 297mm;
            background-image: url("data:image/png;base64...");
            background-repeat: no-repeat;
        }

        @page {
            margin: 0cm;
        }

</style>

<div class="background">

    <table width="100%">
        <tr>
            <td align="left">

            </td>
            <td align="right">
                <p>fsesfsf</p>
                <p>fsk</p>
            </td>
        </tr>
    </table>
</div>
```

**Comment by bsweeney** [comment]
There are some paging concerns with elements that bump up against the bound of the page. Have you tried the most recent release (0.8.4)?

Also, are you trying to span the entire page? In that case you could apply the background image to the body instead.

**Comment by k0nG** [comment]
I'm experiencing the same issue with a `<img>` tag where the image dimensions match the `@page` dimensions.

If I increase the `@page` height by ~3mm it seems to fix it. 

I'm running DOMPDF 1.0.2 on php8.

**Comment by Mellthas** [comment]
This should be fixed by #2506. In the meantime, you can prevent the extra page by ensuring that there is no white space between `</div>` and `</body>`.

**Comment by k0nG** [comment]
Thanks for the tip. I've tried that and it still doesn't seem to work.

It seems when I have an HTML element that is the same or near the same dimensions of the `@page` rule this happens. 

Here's my original HTML:

```html
<!DOCTYPE html >
<html>
<head>
    <title>Page</title>
    <meta http-equiv='Content-Type' content='text/html;charset=UTF-8' />
    

    <style type='text/css'>
        * {
            margin:0;
            padding:0;
            border:0;
            outline:0;
        }
        
            @page {
                size: 184mm  260mm;
            }

            img {
                width: 184mm;
                height: 260mm;
            }
        
    </style>

</head>
<body><img src="http://www.example.com/removeimage.jpg"></body></html>
```

I've also tried using a background image instead. Which oddly then adds an extra blank page **after** the content.

```html
<!DOCTYPE html >
<html>
<head>
    <title>Page</title>
    <meta http-equiv='Content-Type' content='text/html;charset=UTF-8' />
    
    <style type='text/css'>
        * {
            margin:0;
            padding:0;
            border:0;
            outline:0;
        }
        
            @page {
                size: 184mm  260mm;
                
            }
            
            div {
                background-image: url("http://www.example.com/removeimage.jpg");
                background-repeat: no-repeat;
                background-size: 184mm 260mm;
                width: 184mm;
                height: 260mm;
            }
        
    </style>

</head>
<body><div></div></body></html>
```

**Comment by Mellthas** [comment]
For the second example, having no white space between `</div>` and `</body>` produces only one page for me. Adding a break between them produces two pages, but that is fixed with #2506. This matches the behavior of the original report.

The `img` producing an extra blank page before it must have a different cause. I can force it to one page by ensuring that the image height is smaller than the page height, even by a tiny amount, e.g. by setting the image height to `259.99mm`.

**Comment by k0nG** [comment]
Hi, @Mellthas thanks so much for the assistance on this. I've got it working without a blank page.

Really appreciate the fast replies.
